pluginProjects/FileLoaders/SGM2010FastFileLoader/SGM2010FastFileLoaderPlugin.cpp

```diff
@@ -139,14 +139,12 @@ bool SGM2010FastFileLoaderPlugin::load(AMScan *scan, const QString &userDataFold
 	QString scalerFile = "";
 	QFileInfo scalerFileInfo;
 	if(colNames1.contains("scaler_fileOffset")){
-		foreach(QString afp, scan->additionalFilePaths())
```

Comment:
@dretrex I think it is identical. The removed codes are looking for files (actually there will be only one) with "_spectra.dat"  in the name. If it found, then the `scalerFileInfo` is initialized. If nothing is found, (scalerFile is ""), an error will be throw out.
---
With my changes, the function `scan->spectraFilePath()` will return the spectra data file name or "". If the file name is not "", the scalerFileInfo is initialized, otherwise an error will be thrown.